modules/users/client/config/users.client.routes.js

```diff
@@ -149,15 +149,30 @@
             return SettingsService.get();
           },
 
-          profile: function (UserProfilesService, $stateParams) {
+          profile: function (UserProfilesService, $stateParams, $q) {
             return UserProfilesService.get({
               username: $stateParams.username
-            });
+            }).$promise
+              .catch(function (e) {
+
+                if (e.status === 404) {
+                  // when user was not found, resolving with empty user profile, in order to display the User Not Found error.
+                  return { $promise: $q.resolve({ }), $resolved: true };
+                }
+
+                throw e;
+              });
           },
 
           // Contact is loaded only after profile is loaded, because we need the profile ID
           contact: function (ContactByService, profile, Authentication) {
             return profile.$promise.then(function (profile) {
+
+              // when user doesn't exist, no need to load contact
+              if (!profile.id) {
```

Comment:
Shouldn't this be `profile._id` ?

Reply:
Very much indeed. This would not load any contacts of any users. :astonished: